partner_affiliate/models/res_partner.py

```diff
@@ -1,24 +1,156 @@
 # Copyright 2012 Camptocamp SA - Yannick Vaucher
 # Copyright 2018 brain-tec AG - Raul Martin
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# Copyright 2021 Therp BV - <https://therp.nl>.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+"""Support affiliate relations beween partners."""
 
-from odoo import fields, models
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
 
 
 class ResPartner(models.Model):
     """Add relation affiliate_ids."""
 
+    _name = "res.partner"
     _inherit = "res.partner"
 
-    # force "active_test" domain to bypass _search() override
     child_ids = fields.One2many(
-        domain=[("active", "=", True), ("is_company", "=", False)]
+        "res.partner",
+        "parent_id",
+        string="Contacts",
+        domain=[
+            ("active", "=", True),
+            ("is_company", "=", False),
+        ],
     )
 
-    # force "active_test" domain to bypass _search() override
     affiliate_ids = fields.One2many(
         "res.partner",
         "parent_id",
         string="Affiliates",
-        domain=[("active", "=", True), ("is_company", "=", True)],
+        domain=[
+            ("active", "=", True),
+            ("is_company", "=", True),
+        ],
     )
+
+    type = fields.Selection(
+        selection_add=[
+            ("affiliate", "Affiliate"),
+            ("parent_company", "Parent Company"),
+        ],
+        required=True,
+        default="contact",
+        ondelete={
+            "affiliate": "set default",
+            "parent_company": "set default",
+        },
+        compute="_compute_type",
+        store=True
+    )
+
+    # parent_id_domain = fields.Char(
+    #     string="Parent Company Domain",
+    #     compute="_compute_parent_id_domain",
+    #     store=True,
+    # )
+
+    @api.depends("parent_id", "company_type")
+    def _compute_type(self):
+        for partner in self:
+            if partner.company_type == "person":
+                partner.type = "contact"
+            if not partner.parent_id and partner.company_type == "company":
```

Comment:
partner.company_type può essere solo "company"

```suggestion
if partner.company_type == "person":
    partner.type = "contact"
    continue
partner.type = "affiliate" if partner.parent_id else "parent_company"
```